Nitrox.Test/Helper/Serialization/NitroxAutoBinderBase.cs

```diff
@@ -25,32 +30,31 @@ public override TType CreateInstance<TType>(AutoGenerateContext context)
 
         if (type.IsAbstract && type != typeof(Type)) // System.Type is abstract but that is not important to us
         {
-
             if (!subtypesByBaseType.ContainsKey(type))
             {
                 throw new Exception($"{type} has no registered implementing classes.");
             }
 
-            if (subtypesByBaseType[type].Count == 0)
+            Type subtype;
+
+            // Optionals are for some reason double created. The second creation has an Optional<t> at the second position of the type stack.
+            Type[] typeStack = ((Stack<Type>)typeStackProperty.GetValue(context)).ToArray();
+            if (typeStack.Length > 1 && typeStack[1].IsGenericType && typeStack[1].GetGenericTypeDefinition() == typeof(Optional<>))
             {
-                return default;
+                subtype = subtypesByBaseType[type].Peek();
+            }
+            else
+            {
+                subtype = subtypesByBaseType[type].Dequeue();
```

Comment:
This line throw an error for me in `PacketSerializationTest`:
```cs
Test method NitroxModel.Packets.PacketsSerializableTest.PacketSerializationTest threw exception: 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: Queue empty.
    at System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   at System.Collections.Generic.Queue`1.Dequeue()
   at Nitrox.Test.Helper.Serialization.NitroxAutoBinderBase.CreateInstance[TType](AutoGenerateContext context) in Nitrox.Test\Helper\Serialization\NitroxAutoBinderBase.cs:line 48
   at AutoBogus.Generators.TypeGenerator`1.AutoBogus.IAutoGenerator.Generate(AutoGenerateContext context)
   at System.Linq.Enumerable.WhereSelectArrayIterator`2.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at AutoBogus.AutoBinder.CreateInstance[TType](AutoGenerateContext context)
   at Nitrox.Test.Helper.Serialization.NitroxAutoBinderBase.CreateInstance[TType](AutoGenerateContext context) in Nitrox.Test\Helper\Serialization\NitroxAutoBinderBase.cs:line 58
   at AutoBogus.AutoFaker`1.<>c__DisplayClass29_0.<PrepareCreate>b__0(Faker faker)
   at Bogus.Faker`1.Generate(String ruleSets)
--- End of inner exception stack trace ---
    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Nitrox.Test.Helper.Serialization.NitroxAutoFakerNonGeneric.Generate[T](Type type) in Nitrox.Test\Helper\Serialization\NitroxAutoFakerNonGeneric.cs:line 40
   at NitroxModel.Packets.PacketsSerializableTest.PacketSerializationTest() in Nitrox.Test\Model\Packets\PacketsSerializableTest.cs:line 65
```

Reply:
`PacketsSerializableTest` is already broken on master (StackOverflowException) and need to be fixed in another pr imo.